pkg/scheduler/objects/application.go

```diff
@@ -1660,6 +1664,8 @@ func (sa *Application) addAllocationInternal(allocType AllocationResultType, all
 		sa.incUserResourceUsage(alloc.GetAllocatedResource())
 		sa.allocatedPlaceholder = resources.Add(sa.allocatedPlaceholder, alloc.GetAllocatedResource())
 		sa.maxAllocatedResource = resources.ComponentWiseMax(sa.allocatedPlaceholder, sa.maxAllocatedResource)
+		sa.allocatedPlaceholder.Prune()
+		sa.maxAllocatedResource.Prune()
```

Comment:
i would be really surprised if that would ever have a zero in any type. We can only add a type to this aggregate if it is in the request of an allocation as a non zero value. If we prune `allocatedPlaceholder` before we pass it into the calculation we only need one prune call.
move `allocatedPlaceholder.Prune()` upone line and remove `maxAllocatedResource.Prune()`

Reply:
On top of that alloc should never have zero values in it so not sure we need to prune at all here.

Reply:
Thanks @wilfred-s , it makes sense to me, we only need prune for sub subFrom if we add positive value.